### PR TITLE
Module 2 Project Exercise

### PR DESCRIPTION
**Pull request by pat-morris**

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -3,4 +3,11 @@ FLASK_APP=app
...
 
 # Change the following values for local development.
-SECRET_KEY=secret-key
+TRELLO_API_KEY=
+TRELLO_API_TOKEN=
+TRELLO_BOARD_ID=
+TRELLO_TODO_LIST_ID=
+TRELLO_DOING_LIST_ID=
+TRELLO_DONE_LIST_ID=
+TRELLO_BOARDS_URL=
```
Comment (Jonesey13) [inline .env.template:12]: I don't think these last 2 are used

File: requirements.txt [hunk 2]
```diff
@@ -1,2 +1,25 @@
+astroid==2.4.1
```
Comment (Jonesey13) [inline requirements.txt:1]: Do we need all these dependencies?

File: setup.sh [hunk 3]
```diff
@@ -11,7 +11,7 @@ python -m venv env
 
 if [ $Windows == 1 ]
 then source env/scripts/activate
-else source env/bin/activate
+else source env/scripts/activate
```
Comment (Jonesey13) [inline setup.sh]: Do you not like typing `--windows`?

File: templates/index.html [hunk 4]
```diff
@@ -9,26 +9,54 @@ <h1 class="display-4">To-Do App</h1>
...
+      {% for item in items %}
+      <li class="list-group-item">
+        <div class="row">
+          <span class="col"><b>{{ item.title }}</b>: {{ item.status }}</span>
+          <div class="col-auto">
+            <form action="{{ url_for('delete_item', id=item.id) }}" method="post" class="d-inline">
+              <button type="submit" class="btn btn-danger">Delete</button>
+            </form>
+            {% if item.status == 'Not Started' %}
+            <form action="{{ url_for('in_progress_item', id=item.id) }}" method="post" class="d-inline">
```
Comment (Jonesey13) [inline templates/index.html]: I'd indent these one level in to show they're contained within the `if` block

File: trello_items.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        title: The title of the item.
+        description: The description of the item
+
+    Returns:
+        item: The saved item.
+    """
+    api_method = 'POST'
+    requests.request(api_method, trello_cards_url, params={**trello_auth, 'idList': my_todo_list, 'name': title, 'desc': description })
+
+def complete_item(id):
```
Comment (Jonesey13) [inline trello_items.py:44]: I'd be tempted to commonise these 3 PUT methods

File: todo_item_class.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+from trello_config import Config
+class ToDoItem:
+    def __init__(self, id, title, status, description):
+        self.id = id
+        self.title = title
+        self.description = description
+        self.status = status
+
+    @classmethod
+    def fromTrelloCard(cls, card, idList):
```
Comment (Jonesey13) [inline todo_item_class.py:10]: Class method names should probably be snake case: https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

Comment (Jonesey13) [reply todo_item_class.py:10]: However this is a good use of `@classmethod'

File: trello_config.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+"""Trello configuration class."""
+import os
+
+class Config:
+    """Base configuration variables."""
+    TRELLO_API_KEY = os.environ.get('TRELLO_API_KEY')
```
Comment (Jonesey13) [inline trello_config.py:6]: These could become class attributes on a TrelloItems class in `trello_items.py` (avoiding you having to prepend `TRELLO_` to everything here)

**Review by Jonesey13 — APPROVED** [review]
This all looks functionally sound to me!

Left a few minor comments here and there but this is ready to be merged in 👍

File: setup.sh (hunk 3 above)
Comment (pat-morris) [reply setup.sh]: Ha! I've reverted the change back.

File: templates/index.html (hunk 4 above)
Comment (pat-morris) [reply templates/index.html]: Good point! I've indented these by one level

**Pull request merged**